pkg/api/analysisrun/phase.go

```diff
@@ -0,0 +1,14 @@
+package analysisrun
+
+// DynamicPhase represents a way to 'run' a package during its usage lifecycle.
+// This is relevant to dynamic analysis.
+type DynamicPhase string
+
+const (
+	DynamicPhaseImport  DynamicPhase = "import"
```

Comment:
Can we remove the type name prefix here? We're not using it anywhere else so maybe it's worth removing it here too.

Reply:
I'm keen to ensure the constants are namespaced properly. If they were in a package called `dynamicphase` then `dynamicphase.Install` would make sense. Or if they were in a package called `dynamic` then `dynamic.PhaseInstall` would make sense.

But outside of that context, I think the prefix helps with readability. For example:

```
MyFunc(analysisrun.Install)
```

is less clear than

```
MyFunc(analysisrun.DynamicPhaseInstall)
```

There are more characters, but I think the code is more legible.

Go stdlib uses prefixes and suffix in this way. For example:
https://pkg.go.dev/net/http#pkg-constants (Method, Status)
https://pkg.go.dev/io@go1.20.1#pkg-constants (Seek, Err)
https://pkg.go.dev/net@go1.20.1#pkg-variables (IPv4, IPv6)

Reply:
> Or if they were in a package called `dynamic` then `dynamic.PhaseInstall` would make sense.

I like this alternative. Or `dynamic.InstallPhase` which reads a bit more naturally

Reply:
We already have `internal/dynamicanalysis` and `internal/staticanalysis` so I don't think we'd need `internal/dynamic` and `internal/static` on top of those. 

So we could have `internal/dynamicanalysis`, `internal/staticanalysis` and `pkg/.../dynamic`, `pkg/.../static`.

Or, if you'd prefer the external names to be longer, we can rename the internal ones: `internal/dynamic`, `internal/static` and `pkg/.../dynamicanalysis`, `pkg/.../staticanalysis`

Reply:
Lol, I've been thinking of dropping `analysis` from the internal names entirely too, which is why I'm thinking about it.

I'm not convinced yet that `DynamicPhase` should be separated from `analysisrun` at this stage. A package with a single tiny data-type seems unnecessary for the time being.

Reply:
This sounds like something we can do in the future as a follow up cleanup?